sdk/core/src/policies/transport.rs

```diff
@@ -37,6 +50,184 @@ where
 
         let response = { self.http_client.execute_request2(request) };
 
-        Ok(response.await?.into())
+        Ok(response.await?)
+    }
+
+    #[cfg(feature = "mock_transport_consume")]
+    async fn send(
+        &self,
+        _ctx: &mut PipelineContext<C>,
+        request: &mut Request,
+        next: &[Arc<dyn Policy<C>>],
+    ) -> PolicyResult<Response> {
+        use crate::mock_transport::*;
+        // there must be no more policies
+        assert_eq!(0, next.len());
+
+        // deserialize to file both the request and the response
+        let (read_request, read_response) = {
+            let mut request_path = prepare_and_get_transaction_path();
+            let mut response_path = request_path.clone();
+
+            request_path.push(format!("{}_request.json", get_transaction_num()));
+            response_path.push(format!("{}_response.json", get_transaction_num()));
+
+            let mut request_contents_stream = std::fs::File::open(&request_path)
+                .expect(&format!("Cannot open request file {:?}", request_path));
+
+            let mut request = String::new();
+            request_contents_stream
+                .read_to_string(&mut request)
+                .expect("cannot read request file");
+
+            let mut response_contents_stream = std::fs::File::open(&response_path)
+                .expect(&format!("Cannot open response file {:?}", response_path));
+
+            let mut response = String::new();
+            response_contents_stream
+                .read_to_string(&mut response)
+                .expect("cannot read response file");
+
+            (request, response)
+        };
+
+        let read_request: Request =
+            serde_json::from_str(&read_request).expect("error deserializing the request");
+
+        println!("read_request == {:#?}", read_request);
+
+        let read_response: BytesResponse =
+            serde_json::from_str(&read_response).expect("error deserializing the response");
+
+        println!("read_response == {:#?}", read_response);
+
+        // check if the passed request matches the one read from disk
+        // We will ignore some headers that are bound to change every time
+        // We'll probabily want to make the exclusion list dynamic at some point.
+        const SKIPPED_HEADERS: &[&'static str] = &["Date", "x-ms-date", "authorization"];
```

Comment:
These should come from a sanitizer. See the link above for .NET's as an example. It must also be pluggable, as other services - like Cognitive Services - use custom header names for secure tokens.

Reply:
Agreed. I would defer this to another PR to keep things moving though.